### DISCUSSION TITLE
What is the recommended way to make a a custom driver available to all projects?

### DISCUSSION THREAD

**Discussion by AdelQraft** [zephyrproject-rtos/zephyr#65427] [Q&A] [answered]
I am developing some custom drivers for a project. Currently, I am developing them out of tree for this specific project. I would want to make them available globally so I can reuse them in my other projects. What would best way to do that? I know I can probably just move everything to the main Zephyr repo (the one that _west_ created by running `west  init` and `west update`), but I would prefer not to mess around with the main repo. Is there a way I can place them somewhere else on my machine and tell _west_ to also search that path for drivers?

**Comment by faxe1008** [comment]
Hi,

This is exactly what zephyr modules are made for:
[Modules](https://docs.zephyrproject.org/latest/develop/modules.html)
[Manifests](https://docs.zephyrproject.org/latest/develop/west/manifest.html)

**Comment by zafersn** [comment]
@AdelQraft  I think your case is the same as mine. I don't want to mess around with the main zephyr. So I have a folder I called "sdk-mycustomprj" by following topology 3 but for you, topology 2 seems enough, see here topology 2-3 https://docs.zephyrproject.org/latest/develop/west/workspaces.html#west-t2
whenever I have a new driver, board, or custom configuration, I keep all of these in my sdk folder. So, I don't have any problem with my main zephyr folder.
I think it will be helpful for you to create a custom SDK folder and synchronise it with your git. Besides, you will have only your configs and drivers etc.. in your git repo... instead of all zephyr

see this sample 
https://github.com/nrfconnect/ncs-example-application

**Comment by DerekSnell** [comment, marked as answer]
Hi @AdelQraft and @zafersn ,
FYI, the Zephyr Project has a [Zephyr Example Application repo](https://github.com/zephyrproject-rtos/example-application), which shows an example how to do this.  The repo includes an out-of-tree sensor driver.  And the repo includes a manifest file so West can clone the Zephyr repo and other dpendencies when this example repo is initialized.

Best regards

**Comment by huunghiaspkt** [comment]
I have the same question here, I know the [Zephyr Example Application ](https://github.com/zephyrproject-rtos/example-application).
But seem it's a little bit silly for me. Since the example will clone the zephyr repo which is exactlly same as installed in my PC.
Is there away to build and use custom driver without pulling again zephyr repo? 
Or how do I tell west build that "hey, the Zephyr sdk is in **c:\zephyr**, go there and take what ever you need for this project. And I also have a custom driver which is located in ./drivers/sensor/...., please build that cuz I used it in main.c"
